src/backend/logging/log_manager.cpp

```diff
@@ -120,103 +149,108 @@ bool LogManager::EndLogging() {
 // Utility Functions
 //===--------------------------------------------------------------------===//
 
-void LogManager::LogBeginTransaction(cid_t commit_id){
+void LogManager::PrepareLogging() {
   if (this->IsInLoggingMode()) {
+    this->GetBackendLogger();
     auto logger = this->GetBackendLogger();
-    auto record = new TransactionRecord(
-        LOGRECORD_TYPE_TRANSACTION_BEGIN, commit_id);
-    logger->Log(record);
+    int frontend_logger_id = logger->GetFrontendLoggerID();
+    LOG_INFO("Got frontend_logger_id as %d", (int)frontend_logger_id);
+    frontend_loggers[frontend_logger_id]->SetBackendLoggerLoggedCid(*logger);
+  }
+}
+
+void LogManager::DoneLogging() {
+  if (this->IsInLoggingMode()) {
+	auto logger = this->GetBackendLogger();
+	logger->SetLoggingCidLowerBound(INVALID_CID);
   }
 }
 
-void LogManager::LogUpdate(concurrency::Transaction* curr_txn, cid_t commit_id, ItemPointer &old_version, ItemPointer &new_version){
+void LogManager::LogBeginTransaction(cid_t commit_id) {
+  if (this->IsInLoggingMode()) {
+    auto logger = this->GetBackendLogger();
+    TransactionRecord record(LOGRECORD_TYPE_TRANSACTION_BEGIN, commit_id);
+    logger->Log(&record);
+  }
+}
+
+void LogManager::LogUpdate(cid_t commit_id, ItemPointer &old_version,
+                           ItemPointer &new_version) {
```

Comment:
It seems you won't modify variable old_version and new_version. It is better to make them const reference. It confused me that you might want to write into them as return values or side effects of calling this function when I first saw them passed in as non-const reference.